corehq/apps/translations/views.py

```diff
@@ -94,48 +92,36 @@ def download_bulk_ui_translations(request, domain, app_id):
 
 @require_can_edit_apps
 def download_bulk_app_translations(request, domain, app_id):
-    app = get_app(domain, app_id)
-    headers = get_bulk_app_sheet_headers(app)
-    rows = get_bulk_app_sheet_rows(app)
-    temp = io.BytesIO()
-    data = [(k, v) for k, v in six.iteritems(rows)]
-    export_raw(headers, data, temp)
-    filename = '{app_name} v.{app_version} - App Translations'.format(
-        app_name=app.name,
-        app_version=app.version)
-    return export_response(temp, Format.XLS_2007, filename)
-
-
-@require_can_edit_apps
-def download_bulk_multimedia_translations(request, domain, app_id):
     lang = request.GET.get('lang')
     app = get_app(domain, app_id)
-
-    headers = get_bulk_multimedia_sheet_headers(lang)
-    rows = get_bulk_multimedia_sheet_rows(lang, app)
+    headers = get_bulk_app_sheet_headers(app, lang=lang)
+    sheets = get_bulk_app_single_sheet_by_name(app, lang) if lang else get_bulk_app_sheets_by_name(app)
 
     temp = io.BytesIO()
-    export_raw(headers, [(headers[0][0], rows)], temp)
-    filename = '{app_name} v.{app_version} - Multimedia Translations {lang}'.format(
+    data = [(k, v) for k, v in six.iteritems(sheets)]
+    export_raw(headers, data, temp)
+    filename = '{app_name} v.{app_version} - App Translations{lang}'.format(
         app_name=app.name,
         app_version=app.version,
-        lang=lang)
+        lang=' ' + lang if lang else '')
     return export_response(temp, Format.XLS_2007, filename)
 
 
 @no_conflict_require_POST
 @require_can_edit_apps
 @get_file("bulk_upload_file")
 def upload_bulk_app_translations(request, domain, app_id):
-    validate = request.POST.get('validate')
+    lang = request.POST.get('language')
+    validate = request.POST.get('validate') if lang else False
+
     app = get_app(domain, app_id)
     workbook, msgs = get_app_translation_workbook(request.file)
     if workbook:
         if validate:
-            msgs = validate_bulk_app_translation_upload(app, workbook, request.user.email, request.file,
-                                                        request.POST.get('language'))
```

Comment:
i think you removed request.file during master update. It should be added and later run in `validate_bulk_app_translation_upload` if you look at the current state of the method.

Reply:
Are you sure? The way I read the code, `request.file` gets turned into `workbook` a few lines up, and then `workbook` is how the data is passed to `validate_bulk_app_translation_upload`.